src/emqx_ws_connection.erl

```diff
@@ -171,6 +171,7 @@ call(WsPid, Req) when is_pid(WsPid) ->
 
 init(Req, Opts) ->
     %% WS Transport Idle Timeout
+    FailIfNoSubprotocol = proplists:get_value(fail_if_no_subprotocol, Opts),
```

Comment:
To avoid crashing at line 190, it makes more sense to write it like this:
```suggestion
    FailIfNoSubprotocol = proplists:get_bool(fail_if_no_subprotocol, Opts),
```